engine/rotationmanager/advance.go

```diff
@@ -35,6 +38,11 @@ func calcAdvance(t time.Time, rot *rotation.Rotation, state rotation.State, part
 		return nil
 	}
 
+	warningThreshold := time.Minute * -15
+	if !newStart.After(t.Add(warningThreshold)) {
+		log.Debugf(ctx, "unexpected rotation advancement")
```

Comment:
Include the amount of time so the log entry is more useful, and be more specific in the message:
```suggestion
		log.Debugf(ctx, "rotation advanced late (%s)", t.Sub(newStart).String())
```